src/components/Passport.js

```diff
@@ -46,139 +61,256 @@ const buildKeyType = permissions => {
+    const chunks = chunkStr(b10, chunkSize).map(c => new BN(c));
+    // generate an empty matrix
+    const startMatrix = Matrix.new(rows, cols);
+    // do a few random walks starting at diff coordinates
+    const [, matrix] = chunks.reduce((acc, chunk) => {
+      return walk(chunk, acc[1]);
+      // start with the 'random' points (not actually random, they are deterministic)
+    }, rand(big, startMatrix));
+
+    return matrix;
+  };
+
+  const matrix = address.matchWith({
+    Just: ({ value }) => makeMatrix(value),
+    Nothing: defaultMatrix,
   });
 
-  const showTicket = ticket !== null;
-  const isValidTicket = !!ticket;
-  const visibleTicket = isValidTicket ? VALID_TICKET : PENDING_TICKET;
-  const visibleAddress = `${address.matchWith({
-    Nothing: () => EMPTY_ADDRESS,
-    Just: p => p.value.slice(0, VISIBLE_ADDRESS_CHAR_COUNT),
-  })}${ADDRESS_BUFFER_CHARS}`;
+  const bgColor = inverted ? 'white' : 'black';
+
+  const fgColor = inverted ? 'black' : 'white';
 
   return (
-    <Grid onClick={onClick} gap={4} className={cn('bg-black r8 p7', className)}>
-      <Grid.Item cols={[1, 4]}>
-        <MaybeSigil patp={name} size={64} />
-      </Grid.Item>
-      <Grid.Item as={Flex} cols={[4, 13]} col justify="between">
-        <Flex.Item className="mono white f5 pb2">{visualName}</Flex.Item>
-        <Flex.Item as={Flex} col>
-          <Flex.Item as={Text} className="mono f6 f5-ns gray4 uppercase">
-            Birth Time
-          </Flex.Item>
-          <Flex.Item as={Text} className="mono f6 f5-ns gray4 uppercase">
-            {visualBirthday}
-          </Flex.Item>
-        </Flex.Item>
-      </Grid.Item>
-      {showTicket && (
-        <Grid.Item full className="mv3" as={Flex} col>
-          <Flex.Item as={Text} className="mono f6 f5-ns gray4 uppercase">
-            Master Ticket
-          </Flex.Item>
-          <Flex.Item
-            as={Text}
-            className={cn('arial f6 f5-ns wrap', {
-              gray4: !isValidTicket,
-              green3: isValidTicket,
-            })}>
-            {visibleTicket}
-          </Flex.Item>
-          <Flex.Item as={Text} className="mono f6 f5-ns gray4 wrap">
-            {visibleAddress}
-          </Flex.Item>
-        </Grid.Item>
-      )}
-    </Grid>
+    <div
+      style={{
+        fontFamily: 'Inter',
+        backgroundColor: bgColor,
+        border: inverted ? '2px solid #E6E6E6' : '2px solid black',
+        borderRadius: '20px',
+        marginBottom: '16px',
+        minWidth: `${16 + 16 + cols * tile}px`,
+        display: 'flex',
+        flexDirection: 'column',
+      }}>
+      <div
+        style={{
+          height: '64px',
+          padding: '16px',
+          display: 'flex',
+        }}>
+        <div
+          style={{
+            width: '100%',
+            display: 'flex',
+            alignItems: 'center',
+          }}>
+          {Just.hasInstance(point) && (
+            <MaybeSigil patp={patp.map(ob.patp).value} />
```

Comment:
I'm getting "patp.map" is not a function

Reply:
Resolved in 4543574